docs/deployment-patterns/deployment-as-code.md

```diff
@@ -0,0 +1,111 @@
+
+## Project as code {#project-as-code}
+
+The next level of **deployment as code** is to define the configuration of your Octopus **project as code**, primarly the deployment process and variables. You can colocate your Octopus project configuration with your application source code, adding a step to your build process which pushes the configuration changes to your Octopus project.
+
+This approach is more complex since you need to change your build pipeline, and train people how to treat the code as the source of truth for these parts of your project.
+
+### Move your project configuration to source code
+
+Follow this process to move your Octopus project configuration to code:
+
+1. Move your [deployment scripts to code](#scripts-as-code) and decide if you want to take this next step
+1. Convert your project configuration to code - learn about [configuring Octopus using code](#configure-octopus-using-code)
+1. Test your project configuration as code, targeting a dummy project so your real project continues to work uninterrupted
+1. Once you are happy with your testing, update your build pipeline to target the real project
+1. Train your people to make changes safely using **project as code**
+
+The process flow using **project as code** looks similar to what you already have, with on additional step to push the configuration changes into Octopus. Here is a typical example:
+
+1. Build and test application source code
+1. Package application source code
+1. **Push configuration changes to Octopus project treating the code as a source of truth**
+1. Push application packages to deployment feed
+1. Create a release in Octopus
+1. Deploy the release via Octopus to your Dev/Test environment
+1. Promote the release to other environments
+
+### Configure Octopus using code {#configure-octopus-using-code}
+
+Octopus has a comprehensive HTTP API and .NET SDK you can use to automate **everything** in Octopus - if you can do something through the user interface, you can automate it with code. You can create and update projects, variables, deployment processes, and more. Today, this is our only fully-supported **Octopus as Code** solution.
+
+There is an [open source Terraform provider for Octopus](https://github.com/MattHodge/terraform-provider-octopusdeploy), which is built on top of the Octopus HTTP API. We are using this Terraform provider ourselves for [Octopus Cloud](https://octopus.com/cloud) (our SaaS product), and we are actively contributing to it. It doesn’t cover 100% of all Octopus features yet, and the structure of the Terraform resources are subject to change. We will be building first-class support for this into the Octopus ecosystem in the future.
```

Comment:
`and we are actively contributing to it`... unclear which `it` you are talking about.
Consider `and we are actively contributing to the provider`.